source/encoder.c

```diff
@@ -377,21 +382,64 @@ encode_length(CBOREncoderObject *self, const uint8_t major_tag,
     }
 }
 
+static int
+encode_length(CBOREncoderObject *self, const uint8_t major_tag,
+    const uint64_t length) {
+    return encode_length_possibly_indefinite(self, major_tag, length, false);
+}
+
+int uint64_or_none(PyObject *obj, void *param) {
+    if (obj ==Py_None) {
+        return 1;
+
+    } else if (PyLong_Check(obj)) {
+        const uint64_t val = PyLong_AsUnsignedLong(obj);
+        if (PyErr_Occurred()) {
+            return 0;
+        }
+
+        *((uint64_t*)param) = val;
+        return 1;
+
+    } else {
+        PyErr_SetString(PyExc_TypeError, "must be int or None");
+        return 0;
+    }
+}
 
 // CBOREncoder.encode_length(self, major_tag, length)
 static PyObject *
 CBOREncoder_encode_length(CBOREncoderObject *self, PyObject *args)
 {
     uint8_t major_tag;
-    uint64_t length;
+    uint64_t length = -1;
```

Comment:
Yes, -1 becomes a very large integer. 

I did this because I was too lazy to set up a custom structure for the `uint64_or_none` that would be required for `PyArg_ParseTuple`. It would be cleaner to have it distinct, yes. Should I remake it?

Reply:
Well, I just thought that if using -1 was acceptable, why not just use it directly in `encode_length()`?

Reply:
I guess I can - if you accept that solution, I can make it.

Reply:
Do you see any downsides when compared to the current code in this PR?

Reply:
At the current state of the code, probably not.

Reply:
Updated.